crates/kornia-io/src/stream/capture.rs

```diff
@@ -90,21 +103,41 @@ impl StreamCapture {
             .lock()
             .map_err(|_| StreamCaptureError::MutexPoisonError)?;
         if let Some(frame_buffer) = circular_buffer.pop_front() {
-            // TODO: solve the zero copy issue
-            // https://discourse.gstreamer.org/t/zero-copy-video-frames/3856/2
-            let buffer = frame_buffer
+            let width = frame_buffer.width as usize;
+            let height = frame_buffer.height as usize;
+
+            // Create a mapping of the buffer without moving it out of frame_buffer
+            let buffer_map = frame_buffer
                 .buffer
                 .map_readable()
                 .map_err(|_| StreamCaptureError::GetBufferError)?;
-            let img = Image::<u8, 3>::new(
-                ImageSize {
-                    width: frame_buffer.width as usize,
-                    height: frame_buffer.height as usize,
-                },
-                buffer.to_owned(),
-            )
-            .map_err(|_| StreamCaptureError::CreateImageFrameError)?;
-            return Ok(Some(img));
+
+            let frame_data_slice = buffer_map.as_slice();
+            let frame_data_ptr = frame_data_slice.as_ptr();
+
+            let length = frame_data_slice.len();
+            let shape = [height, width, 3];
+            let strides = get_strides_from_shape(shape);
+
+            // Drop the buffer_map as it is a reference of Buffer
+            drop(buffer_map);
+
+            let gst_parent_deallocator = Arc::new(GstParentDeallocator(frame_buffer.buffer));
+
+            let tensor = unsafe {
+                Tensor {
+                    shape,
+                    strides,
+                    storage: TensorStorage::from_raw_parts(
+                        frame_data_ptr,
```

Comment:
then you tell Rust to ignore the dangling pointer and use it as a valid allocated memory.